spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BatchDataReader.java

```diff
@@ -28,21 +28,48 @@
 import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.spark.source.metrics.TaskNumDeletes;
+import org.apache.iceberg.spark.source.metrics.TaskNumSplits;
 import org.apache.spark.rdd.InputFileBlockHolder;
+import org.apache.spark.sql.connector.metric.CustomTaskMetric;
+import org.apache.spark.sql.connector.read.PartitionReader;
 import org.apache.spark.sql.vectorized.ColumnarBatch;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class BatchDataReader extends BaseBatchReader<FileScanTask> {
+class BatchDataReader extends BaseBatchReader<FileScanTask>
```

Comment:
This class was only used as `PartitionReader` in `SparkScan`, where we extended it, implemented `PartitionReader` and called the implementation as `BatchReader`.  After adding a common reader factory, we may have multiple batch readers now. That's why `BatchDataReader` seemed like a more accurate name than `BatchReader`. As there were no other places that used this class, I decided to implement `PartitionReader` directly here.

Any feedback is welcome. See `SparkScan` below for old usage.